src/main/ngapp/accio/src/app/editor/editor.component.ts

```diff
@@ -1,15 +1,63 @@
 import { Component, OnInit } from '@angular/core';
+import { ViewChild, ElementRef } from '@angular/core';
+import { Router, ActivatedRoute } from '@angular/router';
+
+import { imageUrls } from '../images';
 
 @Component({
   selector: 'app-editor',
   templateUrl: './editor.component.html',
   styleUrls: ['./editor.component.css']
 })
 export class EditorComponent implements OnInit {
+  // Define urls within component
+  url;
+
+  // inject canvas from html
+  @ViewChild('canvas', { static: true })
+  canvas: ElementRef<HTMLCanvasElement>; 
+  private ctx: CanvasRenderingContext2D;
+  image = new Image();
 
-  constructor() { }
+  constructor(
+    private route: ActivatedRoute,
+    private router: Router,
+  ) { }
 
-  ngOnInit(): void {
+  ngOnInit() {
+    //returns url user clicked in gallery component
+    this.route.paramMap.subscribe(params => {
+      let index = Number(params.get('imgUrl'));
+      this.checkImageUrl(index);
+      this.url = imageUrls[index];
+    });
+
+    this.ctx = this.canvas.nativeElement.getContext('2d');
+    this.draw();
+  }  
+  
+  //  Checks that route for editor URL is in bounds of array of Urls and is a number
+  //  Redirects to gallery component if not
+  private checkImageUrl(index: number) {
+    if (Number.isNaN(index) || index >= imageUrls.length) {    
```

Comment:
A Number can be negative right?  So I think you'd also need to check < 0.

Reply:
I guess I'm not sure how the index would be a negative number. Because index comes from the url, which changes depending on what image the user clicks on. If an image is rendered on the gallery page but not in the array, then I could see the use for checking the out of array length, which will come into play with user loaded images, but I'm not sure about a negative. It's not a big check either so it might be fine to leave it in, I was just curious what your thoughts were.

Reply:
Yes the index comes from the url but can't the user manually change the number in the url by hand?

Reply:
ah yes, makes sense!